### ISSUE TITLE
[jsx-key]: report on `yield`

### ISSUE THREAD

**Issue by benmosher** [yannickcr/eslint-plugin-react#352]
Seems like `yield <Component x={y}/>` should be reported by `jsx-key` if there is no `key` prop.

This is valid:
```js
for (let [idx, datum] of data.entries()) {
  yield (<MyComponent datum={datum} key={idx} />)
}
```
This is reported:
```js
for (let [idx, datum] of data.entries()) {
  yield (<MyComponent datum={datum} />) // msg: need `key` prop
}
```

I can knock this out, if it makes sense. 

(cc #332 as it will likely conflict with this if completed concurrently.)

**Comment by benmosher** [comment]
Could actually add options on this to avoid #320, something like

```yaml
rules:
  react/jsx-key: [2, {yield: true, arrow:true, array: true}]
```

Which #320 could avoid by setting `arrow:false`. Not perfect, but simple.

**Comment by ljharb** [comment]
I think yielding a component in a generator is a very strange edge case, but it should be handled.
